spec/inject_spec.rb

```diff
@@ -0,0 +1,28 @@
+require 'inject'
+
+describe Array do
```

Comment:
Tests are a little strange. You seem to be testing that the core Ruby inject method produces 6 and then testing whether your method produces 6. Why not test your version directly against the Ruby version?